pkgs/development/python-modules/mesonpep517/default.nix

```diff
@@ -0,0 +1,46 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, meson
+, ninja
+, intreehooks
+, pytoml
+, pythonOlder
+}:
+
+# TODO: offer meson as a Python package so we have dist-info folder.
+
+buildPythonPackage rec {
+  pname = "mesonpep517";
+  version = "0.1.9999994";
+  format = "pyproject";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "b5bcca61024164c4a51d29e6921ea1f756d54197c8f052e4c66a2b8399aa9349";
+  };
+
+  nativeBuildInputs = [ intreehooks  ];
+
+  propagatedBuildInputs = [ pytoml ];
+
+  # Meson tries to detect ninja as well, so we should patch meson as well.
+  # postPatch = ''
+  #   substituteInPlace mesonpep517/buildapi.py \
+  #     --replace "'meson'" "'${meson}/bin/meson'" \
+  #     --replace "'ninja'" "'${ninja}/bin/ninja'"      
+  # '';
+
+  propagatedNativeBuildInputs = [ meson ninja ];
+
+  # This should be propagated as long as meson comes with a hook.
```

Comment:
I am starting to get an increasing dislike of packages *including* a setup hook when you do not want it. Maybe we should split it off, that is into a `meson` and `mesonHook`.